test_roundtrip() failing on my machine

In working on #68, I ran the test suite (`pytest tests`) on my local machine, and `test_roundtrip()` in `test_solve.py` failed:

```
>       np.testing.assert_almost_equal(err, 0)
E       AssertionError: 
E       Arrays are not almost equal to 7 decimals
E        ACTUAL: 0.7073383390219243
E        DESIRED: 0

tests/test_solve.py:80: AssertionError
```
But if I simply run `pytest tests/test_solve.py` it passes:
```
tests/test_solve.py ....                                                                                         [100%]

=============================================== 4 passed in 0.86 seconds ===============================================
```


Despite the test weirdness, it seemed like this change was not related to the `np.load` issue I was working on, so I made the PR and it passed the Travis/Appveyor tests. Any idea on why this test would fail on my local machine (fresh conda env using `requirements.txt` and `requirements-dev.txt` on macOS) but not on the CI?

Hmm, this _is_ strange. Here are full details of my installation and the errors, where running the full suite fails at `pytest/test_solve.py` but passes when calling that file directly with `pytest`. 

# Running `pytest tests` (fails)
```
$ pytest tests
================================================= test session starts ==================================================
platform darwin -- Python 3.7.3, pytest-4.4.1, py-1.8.0, pluggy-0.11.0
rootdir: /Users/dnowacki/projects/UTide, inifile: setup.cfg
plugins: xdist-1.28.0, forked-1.0.2, flake8-1.0.4, cov-2.7.1
collected 14 items                                                                                                     

tests/test_HNL.py ....                                                                                           [ 28%]
tests/test_astron.py .                                                                                           [ 35%]
tests/test_harmonics.py .                                                                                        [ 42%]
tests/test_normalize.py .                                                                                        [ 50%]
tests/test_periodogram.py ..                                                                                     [ 64%]
tests/test_solve.py F...                                                                                         [ 92%]
tests/test_uv.py .                                                                                               [100%]

======================================================= FAILURES =======================================================
____________________________________________________ test_roundtrip ____________________________________________________

    def test_roundtrip():
        """Minimal conversion from simple_utide_test."""
        ts = 735604
        duration = 35
    
        time = np.linspace(ts, ts+duration, 842)
        tref = (time[-1] + time[0]) / 2
    
        const = ut_constants.const
    
        amp = 1.0
        phase = 53
        lat = 45.5
    
        freq_cpd = 24 * const.freq
    
        jj = 48-1  # Python index for M2
    
        arg = 2 * np.pi * (time - tref) * freq_cpd[jj] - np.deg2rad(phase)
        time_series = amp * np.cos(arg)
    
        opts = {
            'constit': 'auto',
            'phase': 'raw',
            'nodal': False,
            'trend': False,
            'method': 'ols',
            'conf_int': 'linear',
            'Rayleigh_min': 0.95,
        }
    
        speed_coef = solve(time, time_series, time_series, lat=lat, **opts)
        elev_coef = solve(time, time_series, lat=lat, **opts)
    
        amp_err = amp - elev_coef['A'][0]
        phase_err = phase - elev_coef['g'][0]
        ts_recon = reconstruct(time, elev_coef).h
    
        # pure smoke testing of reconstruct
        vel = reconstruct(time, speed_coef)
        vel = reconstruct(time, speed_coef, constit=('M2', 'S2'))
        htmp = reconstruct(time, elev_coef, constit=('M2', 'S2'))
        vel = reconstruct(time, speed_coef, min_SNR=3)
        htmp = reconstruct(time, elev_coef, min_SNR=3)
        vel = reconstruct(time, speed_coef, min_PE=10)
        htmp = reconstruct(time, elev_coef, min_PE=10)
        vel = reconstruct(time, speed_coef, min_SNR=0)
        htmp = reconstruct(time, elev_coef, min_SNR=0)
        assert isinstance(vel, Bunch)
        assert isinstance(htmp, Bunch)
    
        # Now the round-trip check, just for the elevation.
        err = np.sqrt(np.mean((time_series-ts_recon)**2))
    
        print(amp_err, phase_err, err)
        print(elev_coef['aux']['reftime'], tref)
        print(elev_coef['aux']['opt'])
    
        np.testing.assert_almost_equal(amp_err, 0)
        np.testing.assert_almost_equal(phase_err, 0)
>       np.testing.assert_almost_equal(err, 0)
E       AssertionError: 
E       Arrays are not almost equal to 7 decimals
E        ACTUAL: 0.7073383390219243
E        DESIRED: 0

tests/test_solve.py:80: AssertionError
------------------------------------------------- Captured stdout call -------------------------------------------------
solve: matrix prep ... solution ... diagnostics ... done.
solve: matrix prep ... solution ... diagnostics ... done.
prep/calcs ... done.
prep/calcs ... done.
prep/calcs ... done.
prep/calcs ... done.
prep/calcs ... done.
prep/calcs ... done.
prep/calcs ... done.
prep/calcs ... done.
prep/calcs ... done.
prep/calcs ... done.
4.904632255886554e-12 6.356515314109856e-11 0.7073383390219243
735621.5 735621.5
RunTimeDisp :                                     True
cnstit      :                                     auto
conf_int    :                                     True
diagnminsnr :                                        2
diagnplots  :                                        0
epoch       :                                   python
equi        :                                     True
gwchlint    :                                    False
gwchnone    :                                     True
infer       :                                     None
inferaprx   :                                        0
linci       :                                     True
lsfrqosmp   :                                        1
method      :                                      ols
newopts     : MC_n         :                           200
Rayleigh_min :                          0.95
conf_int     :                        linear
constit      :                          auto
epoch        :                        python
infer        :                          None
method       :                           ols
nodal        :                         False
phase        :                           raw
robust_kw    : {'weight_function': 'cauchy'}
trend        :                         False
verbose      :                          True
white        :                         False

nodesatlint :                                    False
nodesatnone :                                    False
nodiagn     :                                        0
nodsatlint  :                                        0
nodsatnone  :                                     True
notrend     :                                     True
nrlzn       :                                      200
ordercnstit :                                     None
prefilt     :                                       []
rmin        :                                     0.95
runtimedisp :                                      yyy
tunrdn      :                                        1
twodim      :                                    False
white       :                                    False

=================================================== warnings summary ===================================================
tests/test_solve.py::test_roundtrip
  /Users/dnowacki/projects/UTide/utide/confidence.py:237: RuntimeWarning: invalid value encountered in double_scalars
    varXv = Pvv[c] * varXv / den

tests/test_solve.py::test_roundtrip
  /Users/dnowacki/projects/UTide/utide/confidence.py:238: RuntimeWarning: invalid value encountered in double_scalars
    varYv = Pvv[c] * varYv / den

tests/test_solve.py::test_roundtrip
  /Users/dnowacki/projects/UTide/utide/confidence.py:242: RuntimeWarning: invalid value encountered in sqrt
    np.sqrt(varXu) + 1j * np.sqrt(varXv),

tests/test_solve.py::test_roundtrip
  /Users/dnowacki/projects/UTide/utide/confidence.py:243: RuntimeWarning: invalid value encountered in sqrt
    np.sqrt(varYu) + 1j * np.sqrt(varYv))

tests/test_solve.py::test_roundtrip
  /Users/dnowacki/projects/UTide/utide/confidence.py:225: RuntimeWarning: invalid value encountered in double_scalars
    varXu = Puu[c] * varXu / den

tests/test_solve.py::test_roundtrip
  /Users/dnowacki/projects/UTide/utide/confidence.py:226: RuntimeWarning: invalid value encountered in double_scalars
    varYu = Puu[c] * varYu / den

tests/test_solve.py::test_roundtrip
  /Users/dnowacki/projects/UTide/utide/_reconstruct.py:106: RuntimeWarning: invalid value encountered in greater_equal
    ind = np.logical_and(SNR >= min_SNR, PE >= min_PE)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
=================================== 1 failed, 13 passed, 7 warnings in 3.21 seconds ====================================
```

# Running `pytest tests/test_solve.py` (passes)
```
$ pytest tests/test_solve.py 
================================================= test session starts ==================================================
platform darwin -- Python 3.7.3, pytest-4.4.1, py-1.8.0, pluggy-0.11.0
rootdir: /Users/dnowacki/projects/UTide, inifile: setup.cfg
plugins: xdist-1.28.0, forked-1.0.2, flake8-1.0.4, cov-2.7.1
collected 4 items                                                                                                      

tests/test_solve.py ....                                                                                         [100%]

=============================================== 4 passed in 0.83 seconds ===============================================
```

# Output of `conda list`
```
$ conda list
# packages in environment at /Users/dnowacki/miniconda3/envs/utide:
#
# Name                    Version                   Build  Channel
alabaster                 0.7.12                     py_0    conda-forge
apipkg                    1.5                        py_0    conda-forge
asn1crypto                0.24.0                py37_1003    conda-forge
atomicwrites              1.3.0                      py_0    conda-forge
attrs                     19.1.0                     py_0    conda-forge
babel                     2.6.0                      py_1    conda-forge
bleach                    3.1.0                      py_0    conda-forge
bzip2                     1.0.6             h1de35cc_1002    conda-forge
ca-certificates           2019.3.9             hecc5488_0    conda-forge
certifi                   2019.3.9                 py37_0    conda-forge
cffi                      1.12.3           py37hccf1714_0    conda-forge
chardet                   3.0.4                 py37_1003    conda-forge
cmarkgfm                  0.4.2            py37h1de35cc_2    conda-forge
coverage                  4.5.3            py37h1de35cc_0    conda-forge
cryptography              2.6.1            py37h212c5bf_0    conda-forge
cycler                    0.10.0                     py_1    conda-forge
docutils                  0.14                  py37_1001    conda-forge
entrypoints               0.3                   py37_1000    conda-forge
execnet                   1.6.0                      py_0    conda-forge
flake8                    3.7.7                    py37_0    conda-forge
flake8-builtins           1.4.1                      py_0    conda-forge
flake8-comprehensions     2.1.0                      py_0    conda-forge
flake8-mutable            1.2.0                      py_1    conda-forge
flake8-print              3.1.0                      py_0    conda-forge
freetype                  2.10.0               h24853df_0    conda-forge
future                    0.17.1                py37_1000    conda-forge
idna                      2.8                   py37_1000    conda-forge
imagesize                 1.1.0                      py_0    conda-forge
jinja2                    2.10.1                     py_0    conda-forge
kiwisolver                1.1.0            py37h770b8ee_0    conda-forge
libblas                   3.8.0                8_openblas    conda-forge
libcblas                  3.8.0                8_openblas    conda-forge
libcxx                    8.0.0                         2    conda-forge
libcxxabi                 8.0.0                         2    conda-forge
libffi                    3.2.1             h6de7cb9_1006    conda-forge
libgfortran               3.0.1                         0    conda-forge
liblapack                 3.8.0                8_openblas    conda-forge
libpng                    1.6.37               h2573ce8_0    conda-forge
markupsafe                1.1.1            py37h1de35cc_0    conda-forge
matplotlib                3.0.3                    py37_1    conda-forge
matplotlib-base           3.0.3            py37habd5b94_1    conda-forge
mccabe                    0.6.1                      py_1    conda-forge
more-itertools            4.3.0                 py37_1000    conda-forge
ncurses                   6.1               h0a44026_1002    conda-forge
numpy                     1.16.3           py37hdf140aa_0    conda-forge
openblas                  0.3.6                hd44dcd8_1    conda-forge
openssl                   1.1.1b               h01d97ff_2    conda-forge
packaging                 19.0                       py_0    conda-forge
pandas                    0.24.2           py37h0a44026_0    conda-forge
pip                       19.1                     py37_0    conda-forge
pkginfo                   1.5.0.1                    py_0    conda-forge
pluggy                    0.11.0                     py_0    conda-forge
py                        1.8.0                      py_0    conda-forge
pycodestyle               2.5.0                      py_0    conda-forge
pycparser                 2.19                     py37_1    conda-forge
pyflakes                  2.1.1                      py_0    conda-forge
pygments                  2.3.1                      py_0    conda-forge
pyopenssl                 19.0.0                   py37_0    conda-forge
pyparsing                 2.4.0                      py_0    conda-forge
pysocks                   1.6.8                 py37_1002    conda-forge
pytest                    4.4.1                    py37_0    conda-forge
pytest-cov                2.7.1                      py_0    conda-forge
pytest-flake8             1.0.4                    py37_0    conda-forge
pytest-forked             1.0.2                      py_0    conda-forge
pytest-xdist              1.28.0                     py_0    conda-forge
python                    3.7.3                h0d93f26_0    conda-forge
python-dateutil           2.8.0                      py_0    conda-forge
pytz                      2019.1                     py_0    conda-forge
readline                  7.0               hcfe32e1_1001    conda-forge
readme_renderer           24.0                       py_0    conda-forge
requests                  2.21.0                py37_1000    conda-forge
requests-toolbelt         0.9.1                      py_0    conda-forge
scipy                     1.2.1                    pypi_0    pypi
setuptools                41.0.1                   py37_0    conda-forge
six                       1.12.0                py37_1000    conda-forge
snowballstemmer           1.2.1                      py_1    conda-forge
sphinx                    2.0.1                      py_0    conda-forge
sphinxcontrib-applehelp   1.0.1                      py_0    conda-forge
sphinxcontrib-devhelp     1.0.1                      py_0    conda-forge
sphinxcontrib-htmlhelp    1.0.2                      py_0    conda-forge
sphinxcontrib-jsmath      1.0.1                      py_0    conda-forge
sphinxcontrib-qthelp      1.0.2                      py_0    conda-forge
sphinxcontrib-serializinghtml 1.1.1                      py_0    conda-forge
sqlite                    3.26.0            h1765d9f_1001    conda-forge
tk                        8.6.9             ha441bb4_1001    conda-forge
tornado                   6.0.2            py37h01d97ff_0    conda-forge
tqdm                      4.31.1                     py_0    conda-forge
twine                     1.13.0                     py_1    conda-forge
urllib3                   1.24.2                   py37_0    conda-forge
utide                     0.2.3+2.g5ad0aba           dev_0    <develop>
webencodings              0.5.1                      py_1    conda-forge
wheel                     0.33.1                   py37_0    conda-forge
xz                        5.2.4             h1de35cc_1001    conda-forge
zlib                      1.2.11            h1de35cc_1004    conda-forge
```

--- a/tests/test_solve.py
+++ b/tests/test_solve.py
@@ -15,29 +15,35 @@ from utide._ut_constants import ut_constants
 from utide.utilities import Bunch
 
 
+ts = 735604
+duration = 35
+
+time = np.linspace(ts, ts + duration, 842)
+tref = (time[-1] + time[0]) / 2
+
+const = ut_constants.const
+
+amp = 1.0
+phase = 53
+lat = 45.5
+
+freq_cpd = 24 * const.freq
+
+jj = 48 - 1  # Python index for M2
+
+arg = 2 * np.pi * (time - tref) * freq_cpd[jj] - np.deg2rad(phase)
+tide = amp * np.cos(arg)
+
+np.random.seed(123454321)
+noise = 1e-5 * np.random.randn(len(time))
+
+time_series = tide + noise
+
 # We omit the 'MC' case for now because with this test data, it
 # fails with 'numpy.linalg.LinAlgError: SVD did not converge'.
 @pytest.mark.parametrize("conf_int", ["linear", "none"])
 def test_roundtrip(conf_int):
     """Minimal conversion from simple_utide_test."""
-    ts = 735604
-    duration = 35
-
-    time = np.linspace(ts, ts + duration, 842)
-    tref = (time[-1] + time[0]) / 2
-
-    const = ut_constants.const
-
-    amp = 1.0
-    phase = 53
-    lat = 45.5
-
-    freq_cpd = 24 * const.freq
-
-    jj = 48 - 1  # Python index for M2
-
-    arg = 2 * np.pi * (time - tref) * freq_cpd[jj] - np.deg2rad(phase)
-    time_series = amp * np.cos(arg)
 
     opts = {
         "constit": "auto",
@@ -70,37 +76,19 @@ def test_roundtrip(conf_int):
     assert isinstance(htmp, Bunch)
 
     # Now the round-trip check, just for the elevation.
-    err = np.sqrt(np.mean((time_series - ts_recon) ** 2))
+    err = np.sqrt(np.mean((tide - ts_recon) ** 2))
 
     print(amp_err, phase_err, err)
     print(elev_coef["aux"]["reftime"], tref)
     print(elev_coef["aux"]["opt"])
 
-    np.testing.assert_almost_equal(amp_err, 0)
-    np.testing.assert_almost_equal(phase_err, 0)
-    np.testing.assert_almost_equal(err, 0)
+    np.testing.assert_almost_equal(amp_err, 0, decimal=4)
+    np.testing.assert_almost_equal(phase_err, 0, decimal=4)
+    np.testing.assert_almost_equal(err, 0, decimal=4)
 
 
 def test_masked_input():
     """Masked values in time and/or time series."""
-    ts = 735604
-    duration = 35
-
-    time = np.linspace(ts, ts + duration, 842)
-    tref = (time[-1] + time[0]) / 2
-
-    const = ut_constants.const
-
-    amp = 1.0
-    phase = 53
-    lat = 45.5
-
-    freq_cpd = 24 * const.freq
-
-    jj = 48 - 1  # Python index for M2
-
-    arg = 2 * np.pi * (time - tref) * freq_cpd[jj] - np.deg2rad(phase)
-    time_series = amp * np.cos(arg)
 
     opts = {
         "constit": "auto",
@@ -133,8 +121,8 @@ def test_masked_input():
     elev = reconstruct(time, elev_coef)
     assert isinstance(elev, Bunch)
 
-    np.testing.assert_almost_equal(amp_err, 0)
-    np.testing.assert_almost_equal(phase_err, 0)
+    np.testing.assert_almost_equal(amp_err, 0, decimal=4)
+    np.testing.assert_almost_equal(phase_err, 0, decimal=4)
 
 
 def test_robust():
@@ -146,32 +134,14 @@ def test_robust():
     Minimal conversion from simple_utide_test
 
     """
-    ts = 735604
-    duration = 35
-
-    time = np.linspace(ts, ts + duration, 842)
-    tref = (time[-1] + time[0]) / 2
-
-    const = ut_constants.const
-
-    amp = 1.0
-    phase = 53
-    lat = 45.5
-
-    freq_cpd = 24 * const.freq
-
-    jj = 48 - 1  # Python index for M2
-
-    arg = 2 * np.pi * (time - tref) * freq_cpd[jj] - np.deg2rad(phase)
-    time_series = amp * np.cos(arg)
 
     # Add noise
-    np.random.seed(1)
-    time_series += 0.01 * np.random.randn(len(time_series))
+    np.random.seed(13579)
+    noisy = tide + 0.01 * np.random.randn(len(time))
 
     # Add wild points
-    time_series[:5] = 10
-    time_series[-5:] = -10
+    noisy[:5] = 10
+    noisy[-5:] = -10
 
     opts = {
         "constit": "auto",
@@ -183,36 +153,21 @@ def test_robust():
         "Rayleigh_min": 0.95,
     }
 
-    speed_coef = solve(time, time_series, time_series, lat=lat, **opts)
-    elev_coef = solve(time, time_series, lat=lat, **opts)
+    speed_coef = solve(time, noisy, noisy, lat=lat, **opts)
+    elev_coef = solve(time, noisy, lat=lat, **opts)
 
     print(speed_coef.weights, elev_coef.weights)
     print(speed_coef.rf, elev_coef.rf)
 
+    ts_recon = reconstruct(time, elev_coef).h
+    err = np.std(tide - ts_recon)
+    np.testing.assert_almost_equal(err, 0, decimal=2)
+
 
 def test_MC():
-    ts = 735604
-    duration = 35
-
-    time = np.linspace(ts, ts + duration, 842)
-    tref = (time[-1] + time[0]) / 2
-
-    const = ut_constants.const
-
-    amp = 1.0
-    phase = 53
-    lat = 45.5
-
-    freq_cpd = 24 * const.freq
-
-    jj = 48 - 1  # Python index for M2
-
-    arg = 2 * np.pi * (time - tref) * freq_cpd[jj] - np.deg2rad(phase)
-    time_series = amp * np.cos(arg)
-
     # Add noise
     np.random.seed(1)
-    time_series += 0.01 * np.random.randn(len(time_series))
+    noisy = tide + 0.01 * np.random.randn(len(time))
 
     opts = {
         "constit": "auto",
@@ -225,8 +180,8 @@ def test_MC():
         "Rayleigh_min": 0.95,
     }
 
-    speed_coef = solve(time, time_series, time_series, lat=lat, **opts)
-    elev_coef = solve(time, time_series, lat=lat, **opts)
+    speed_coef = solve(time, noisy, noisy, lat=lat, **opts)
+    elev_coef = solve(time, noisy, lat=lat, **opts)
 
     for name, AA, AA_ci, gg, gg_ci in zip(
         elev_coef.name, elev_coef.A, elev_coef.A_ci, elev_coef.g, elev_coef.g_ci

